backend/app/services/external/aws/s3.py

```
@@ -0,0 +1,50 @@
+    def generate_presigned_url(self, file_type: str, expiration: int = 3600):
+        #"""Generate a pre-signed URL for uploading a file to S3."""
+        extension = guess_extension(file_type)
+        s3_file_key = file_key(extension=f"{extension}")
+        try:
+            presigned_url = self.s3_client.generate_presigned_url('put_object',
+                                                                    Params={'Bucket': self.bucket_name,
+                                                                        'Key': s3_file_key,
+                                                                        'ContentType': file_type},
+                                                                    ExpiresIn=expiration)
+            logger.info(presigned_url)
+            return {"presigned_url": presigned_url, "key": s3_file_key, "bucket": self.bucket_name}
```

Ruff (0.14.10)

Consider moving this statement to an `else` block

(TRY300)